oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/VersionGarbageCollector.java

```diff
@@ -924,6 +930,7 @@ public DetailedGC(@NotNull RevisionVector headRevision, long toModifiedMs, @NotN
             this.timer = createUnstarted();
             // clusterId is not used
             this.revisionForModified = Revision.newRevision(0);
+            this.ownHeadRevision = headRevision.getRevision(nodeStore.getClusterId());
```

Comment:
Why do we need different headRevisions here? We already have a headRevision for the whole root/tree.

Reply:
One is a RevisionVector, the other is just the Revision of the own clusterId of the same headRevision. It is needed for `getNodeAtRevision`